src/compiler/utilities.ts

```diff
@@ -2866,4 +2870,6 @@ namespace ts {
     export function isParameterPropertyDeclaration(node: ParameterDeclaration): boolean {
         return node.flags & NodeFlags.AccessibilityModifier && node.parent.kind === SyntaxKind.Constructor && isClassLike(node.parent.parent);
     }
+
+    export const maxProgramSize = 20 * 1024 * 1024;

```

Comment:
Changed the upper limit from 35M to 20M, as on my machine even 35M would still run out of memory